File: examples/chapter09/baryexample.m
%%
% We use $n=3$ and $n=6$ equally spaced nodes for the function
% $\sin(e^{2x})$ over $[0,1]$.

%%  
f = @(x) sin(exp(2*x));
x = linspace(0,1,1001)';
plot(x,f(x)), hold on

t = linspace(0,1,4)'; 
p = polyinterp(t,f(t));
plot(x,p(x))

t = linspace(0,1,7)'; 
p = polyinterp(t,f(t));
plot(x,p(x));

%%
% The curves always intersect at the interpolation nodes. 

File: julia/scripts/chapter09/baryexample.jl

f = x -> sin(exp(2*x))

using Plots
plot(f,0,1,label="function",legend=:bottomleft)

t = (0:3)/3 
y = f.(t)
include("../FNC.jl")
p = FNC.polyinterp(t,y)
scatter!(t,y,color=:black)
plot!(p,0,1,label="interpolant",
    title="Interpolation on 4 nodes")

plot(f,0,1,label="function",legend=:bottomleft)
t = (0:6)/6 
y = f.(t)
p = FNC.polyinterp(t,y)
scatter!(t,y,color=:black)
plot!(p,0,1,label="interpolant",title="Interpolation on 7 nodes")
